### PR TITLE
Updated GUID regeneration for SDK

### PR DESCRIPTION
**Pull request by SimonDarksideJ**
## Overview

Updated GUID regeneration for SDK

## Changes:

- Fixes: #172

## Dependant on

- https://github.com/XRTK/XRTK-Core/pull/569

### CODE REVIEW

File: XRTK.SDK/Packages/com.xrtk.sdk/Editor/SDKPackageInstaller.cs [hunk 1]
```diff
@@ -36,15 +37,23 @@ private static void ImportPackageAssets()
 
         private static void CheckPackage()
         {
+            var updateGUIDs = false;
 
             if (!EditorPreferences.Get($"{nameof(SDKPackageInstaller)}.Profiles", false))
             {
-                EditorPreferences.Set($"{nameof(SDKPackageInstaller)}.Profiles", PackageInstaller.TryInstallAssets(HiddenProfilePath, $"{DefaultPath}\\Profiles"));
+                EditorPreferences.Set($"{nameof(SDKPackageInstaller)}.Profiles", PackageInstaller.TryInstallAssets(HiddenProfilePath, $"{DefaultPath}\\Profiles", false));
+                updateGUIDs = true;
```
Comment (StephenHodgson) [inline XRTK.SDK/Packages/com.xrtk.sdk/Editor/SDKPackageInstaller.cs:45]: should we break out the TryInstallAssets to set the flag, then pass the flag into the pref setter?

I'm not sure we always want to regenerate guids if installation fails for any reason.

Comment (SimonDarksideJ) [reply XRTK.SDK/Packages/com.xrtk.sdk/Editor/SDKPackageInstaller.cs]: Updated, but still not sure I'm happy with it

Comment (StephenHodgson) [reply XRTK.SDK/Packages/com.xrtk.sdk/Editor/SDKPackageInstaller.cs]: Yeah I agree. Wanna play with it some more and see what you think?

File: XRTK.SDK/Packages/com.xrtk.sdk/Editor/SDKPackageInstaller.cs [hunk 2]
```diff
@@ -36,15 +37,30 @@ private static void ImportPackageAssets()
...
+
+            if (updateProfileGUIDs && updatePrefabsGUIDs)
+            {
+                PackageInstaller.RegenerateGuids(new[] { $"{DefaultPath}\\Profiles", $"{DefaultPath}\\Prefabs" });
+            }
+            else
+            {
+                //Unset prefs as one of the copies has failed.
+                EditorPreferences.Set($"{nameof(SDKPackageInstaller)}.Profiles", false);
+                EditorPreferences.Set($"{nameof(SDKPackageInstaller)}.Prefabs", false);
```
Comment (StephenHodgson) [inline XRTK.SDK/Packages/com.xrtk.sdk/Editor/SDKPackageInstaller.cs:63]: This re-wrote all my guids just by opening the project without a key set. People may run into this same issue when they clone the repo the first time.

Comment (StephenHodgson) [reply XRTK.SDK/Packages/com.xrtk.sdk/Editor/SDKPackageInstaller.cs:63]: https://github.com/XRTK/SDK/pull/174#discussion_r420257430

**Pull request merged**